### PR TITLE
Organization dash cards tests migration (fixes : #2806)

### LINKED ISSUE #2806: Refactor `src/components/OrganizationDashCards/CardItem.test.tsx` from Jest to Vitest
**Is your feature request related to a problem? Please describe.**

This files in this directory and all its subdirectories are currently tested using Jest. As part of our migration to Vitest, we need to refactor all these files' test cases to align with Vitest's syntax and features.

Files:
1. Refactor `src/components/OrganizationDashCards/CardItem.test.tsx` from Jest to Vitest
1. Refactor `src/components/OrganizationDashCards/DashboardCard.test.tsx` from Jest to Vitest

**Acceptance Criteria**

For each `*.test.ts` and `*.test.tsx` file found under the root:

1. Replace Jest-specific functions and mocks with Vitest equivalents
1. Rename the test file from having a `.test.*` suffix to  a `.spec.*` suffix
1. Ensure all tests in the file pass after migration using `npm run test:vitest`
1. Maintain the test coverage for the file as 100% after migration
1. Upload a video or photo for this specific file coverage is 100% in the PR description

**Additional context**

- Refer to the merged PR for the base Vitest configuration:
https://github.com/PalisadoesFoundation/talawa-admin/pull/2457/files#diff-9a5a35cf8d7f02b2bba90905714b762560579c4789e9eb4cebb4607980aa3d93
- Vitest documentation:
https://vitest.dev/

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by amaan-aly246**
**What kind of change does this PR introduce?**

Migration of test cases written in Jest to Vitest and merged CSS files into `app.module.css`

**Issue Number:**

Fixes #2806 

**Did you add tests for your changes?**

Yes, additional test cases have been added for components that did not have test cases written in Jest.

**Snapshots/Videos:**

Screenshot 2024-12-25 235223
Screenshot 2024-12-26 064126

**If relevant, did you update the documentation?**

No

**Does this PR introduce a breaking change?**
No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Introduced new styles for card components to enhance layout and appearance, including responsive design adjustments for smaller screens.
	- Added loading states for card components with new test cases to ensure proper rendering.

- **Bug Fixes**
	- Updated import paths for styles to ensure consistent application of styles across card components.

- **Tests**
	- Added new test suites for `CardItemLoading` and `DashBoardCardLoading` components to verify rendering and child elements.

- **Chores**
	- Removed outdated CSS modules that are no longer in use.

### CODE REVIEW

**Review by palisadoes — DISMISSED** [review]

**Comment by palisadoes** [comment]
Please resolve the conflicting file

**Comment by amaan-aly246** [comment]
@palisadoes It's showing me that this branch is out-of-date with the `develop-postgres` branch. I'm closing this PR and will raise the new PR from the updated base branch.

**Comment by palisadoes** [comment]
Reopening. You don't have to do this. Just update your local branch with the latest upstream.

**Comment by palisadoes** [comment]
Please make coderabbit.ai approve your PR

File: src/style/app.module.css [hunk 1]
```diff
@@ -2274,31 +2334,97 @@ button[data-testid='createPostBtn'] {
...
+  color: var(--bs-primary);
+  overflow: hidden;
+  display: -webkit-box;
+  -webkit-line-clamp: 1;
+  line-clamp: 1;
+  -webkit-box-orient: vertical;
+}
 
-  100% {
-    opacity: 1;
```
Comment (palisadoes) [inline src/style/app.module.css:2300]: Why was this deleted? Won't it affect other layouts?

File: src/style/app.module.css [hunk 2]
```diff
@@ -2274,31 +2334,97 @@ button[data-testid='createPostBtn'] {
...
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  opacity: 0.12;
+  border-radius: 50%;
 }
 
-@keyframes zoomIn {
-  0% {
```
Comment (palisadoes) [inline src/style/app.module.css:2293]: Why was this deleted? Won't it affect other layouts?

File: src/style/app.module.css [hunk 3]
```diff
@@ -2274,31 +2334,97 @@ button[data-testid='createPostBtn'] {
...
+  height: 40px;
+  width: 40px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
 
-  100% {
-    opacity: 1;
-    -webkit-transform: scale(1);
```
Comment (palisadoes) [inline src/style/app.module.css:2287]: Why was this deleted? Won't it affect other layouts?

File: src/style/app.module.css [hunk 4]
```diff
@@ -2274,31 +2334,97 @@ button[data-testid='createPostBtn'] {
...
+  position: relative;
+  display: flex;
+  align-items: center;
+  border: 1px solid var(--bs-gray-200);
+  border-radius: 8px;
+  margin-top: 20px;
+}
 
-@-webkit-keyframes zoomIn {
-  0% {
```
Comment (palisadoes) [inline src/style/app.module.css:2279]: Why was this deleted? Won't it affect other layouts?

**Review by palisadoes — CHANGES_REQUESTED** [review]
1. See comments
2. You have not deleted the `*.test.*` files as requested in the PR

**Comment by amaan-aly246** [comment]
> 1. See comments
> 2. You have not deleted the `*.test.*` files as requested in the PR

@palisadoes all `*.test.*` files are deleted. 
Screenshot 2024-12-26 213832

**Comment by palisadoes** [comment]
> > 1. See comments
> > 2. You have not deleted the `*.test.*` files as requested in the PR
> 
> @palisadoes all `*.test.*` files are deleted. Screenshot 2024-12-26 213832

Thanks, please address the other questions

**Comment by amaan-aly246** [comment]
> Thanks, please address the other questions
I have requested your review on those unresolved conversations. 
Apologies for taking so much time.

**Comment by palisadoes** [comment]
I don't understand. You haven't responded to my queries about the deleted sections in the dialog boxes.

**Comment by amaan-aly246** [comment]
@palisadoes I have replied to you in those dialog boxes. I mentioned there that CodeRabbit requested the deletion of those sections because they were duplicates, in order to resolve the conversation.
The code, even after removing those sections, still passes all the test cases (if that helps). 
Screenshot 2024-12-26 224650

**Comment by amaan-aly246** [comment]
@palisadoes Don't know what's happening , I tried to resolved this issue using `npm run format:fix` , even locally it corrected it , pushed the code but still failing the format check.
Now my local branch is behind or something.

Issue: 
Screenshot 2024-12-26 235657
Locally : 
Screenshot 2024-12-26 235927

**Comment by amaan-aly246** [comment]
Suggest me what should I do now.

**Comment by palisadoes** [comment]
Please update your local branch with the latest upstream. That should fix the issue. If not, please ask the #talawa-admin channel in slack

**Review by palisadoes — APPROVED** [review]

**Pull request merged**